LeaderboardBackend/Models/Entities/ApplicationContext.cs

```diff
@@ -61,4 +90,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         modelBuilder.HasCollation(CASE_INSENSITIVE_COLLATION, "und-u-ks-level2", "icu", deterministic: false);
         modelBuilder.HasPostgresEnum<UserRole>();
     }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder opt)
+    {
+        opt.UseNpgsql(_dataSourceProvider.Value, o => o.UseNodaTime());
+        opt.UseSnakeCaseNamingConvention();
+    }
+}
+
+public class AppContextDataSourceProvider
+{
+    private static int _cacheKey;
+    private static NpgsqlDataSource? _cachedDataSource;
```

Comment:
Shouldn't need to use statics as long as you do AddSingleton during startup, which you do.

Reply:
It is needed (tried without statics before).
The issue with "More than twenty 'IServiceProvider' instances have been created for internal use by Entity Framework" happened only in tests because WebApplicationFactory is built multiple times, so each would get their own unique singleton.
Entity framework's error is based on a static cache.

(on this branch I have to duplicate test files in order to reach the "more than 20" threshold to properly reproduce the issue)

Reply:
I see. I wonder if we're doing something wrong then.